databroker/src/grpc/kuksa_val_v2/val.rs

```diff
@@ -660,43 +734,679 @@ mod tests {
+                    panic!("Should not get Ok as signal test.datapoint1 should not exist")
                 }
+                if config.request_second & !config.auth_second {
+                    panic!("Should not get Ok as we do not have permission for signal test.datapoint2 ")
+                }
+
+                let get_response = response.into_inner();
+
+                let mut response_signals = Vec::new();
+
+                if config.request_first {
+                    let value = proto::Value {
+                        typed_value: Some(proto::value::TypedValue::Int32(-64)),
+                    };
+                    let datapoint = proto::Datapoint {
+                        timestamp: Some(timestamp.into()),
+                        value: Some(value),
+                    };
+                    response_signals.push(datapoint);
+                }
+                if config.request_second {
+                    let value = proto::Value {
+                        typed_value: Some(proto::value::TypedValue::Int32(-13)),
+                    };
+                    let datapoint = proto::Datapoint {
+                        timestamp: Some(timestamp.into()),
+                        value: Some(value),
+                    };
+                    response_signals.push(datapoint);
+                }
+
+                assert_eq!(
+                    get_response,
+                    proto::GetValuesResponse {
+                        data_points: response_signals,
+                    }
+                );
             }
-            Err(err) => {
-                assert!(f, "Error {:?}", err)
+            Err(status) => {
+                // It can be discussed what has precendce NotFound or Unauthenticated, does not really matter
+                // For now assuming that NotFound has precedence, at least if we have a valid token
+                if !config.send_auth {
+                    assert_eq!(status.code(), tonic::Code::Unauthenticated)
+                } else if config.request_first & !config.first_exist {
+                    assert_eq!(status.code(), tonic::Code::NotFound)
+                } else if config.request_first & !config.auth_first {
+                    assert_eq!(status.code(), tonic::Code::PermissionDenied)
+                } else if config.request_second & !config.second_exist {
+                    assert_eq!(status.code(), tonic::Code::NotFound)
+                } else if config.request_second & !config.auth_second {
+                    assert_eq!(status.code(), tonic::Code::PermissionDenied)
+                } else {
+                    panic!("GetValues failed with status: {:?}", status);
+                }
             }
         }
     }
 
+    #[tokio::test]
+    async fn test_get_values_id_one_signal_ok() {
+        let config = GetValuesConfigBuilder::new()
+            .first_exist()
+            .request_first()
+            .send_auth()
+            .auth_first()
```

Comment:
If `auth_first` or `auth_second` are set to `True` should not be `send_auth` `True` by default?

Reply:
Well, at least there is no reason to have auth_first as True if you do not intend to send it. There are generally four combos I check:

- We do not send credentials at all
- We send credentials but they are empty
- We send credentials for signal 1
- We send credentials for signal 2
- We send credentials for signal 1 and 2

I can change so that `send_auth` becomes optional if you call one of the other

Reply:
Maybe then leave it as it is, is not a hurdle to call `send_auth` if `auth_first` or `auth_second` are set to `True`

Reply:
You changed it already, thanks